From e6056268c25be8cb406144d9edd78f1052fe9ce0 Mon Sep 17 00:00:00 2001
From: Federico Simoncelli <fsimonce@redhat.com>
Date: Fri, 23 Mar 2012 12:12:49 -0300
Subject: [RHEL6 qemu-kvm PATCH 4/9] Group snapshot: Fix format name for backing file

RH-Author: Federico Simoncelli <fsimonce@redhat.com>
Message-id: <1332504778-17403-5-git-send-email-fsimonce@redhat.com>
Patchwork-id: 38943
O-Subject: [RHEL6.3 qemu-kvm PATCH v6 04/13] Group snapshot: Fix format name for backing file
Bugzilla: 802284
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>
RH-Acked-by: Jeffrey Cody <jcody@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>

From: Paolo Bonzini <pbonzini@redhat.com>

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
Signed-off-by: Kevin Wolf <kwolf@redhat.com>

BZ: 802284

(cherry picked from commit d0895d6e390d59d3dc6edab771335adfea263029)

Deviations from upstream:
 - check if old_bs is inserted

Signed-off-by: Federico Simoncelli <fsimonce@redhat.com>
---
 blockdev.c |    8 +++++++-
 1 files changed, 7 insertions(+), 1 deletions(-)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 blockdev.c |    8 +++++++-
 1 files changed, 7 insertions(+), 1 deletions(-)

diff --git a/blockdev.c b/blockdev.c
index 72a2e14..d571c4b 100644
--- a/blockdev.c
+++ b/blockdev.c
@@ -783,10 +783,16 @@ void qmp_blockdev_group_snapshot_sync(SnapshotDevList *dev_list,
             goto delete_and_fail;
         }
 
+        if (!bdrv_is_inserted(states->old_bs)) {
+            error_set(errp, QERR_DEVICE_HAS_NO_MEDIUM, dev_info->device);
+            goto delete_and_fail;
+        }
+
         /* create new image w/backing file */
         ret = bdrv_img_create(snapshot_file, format,
                               states->old_bs->filename,
-                              drv->format_name, NULL, -1, flags);
+                              states->old_bs->drv->format_name,
+                              NULL, -1, flags);
         if (ret) {
             error_set(errp, QERR_OPEN_FILE_FAILED, snapshot_file);
             goto delete_and_fail;
-- 
1.7.3.2

